metricbeat/mb/module/connector.go

```diff
@@ -73,3 +80,28 @@ func (c *Connector) Connect() (beat.Client, error) {
 		},
 	})
 }
+
+// processorsForConfig assembles the Processors for a Connector.
+func processorsForConfig(
+	beatInfo beat.Info, config connectorConfig,
+) (*processors.Processors, error) {
+	procs := processors.NewList(nil)
+	if !config.Index.IsEmpty() {
+		staticFields := fmtstr.FieldsForBeat(beatInfo.Beat, beatInfo.Version)
+		timestampFormat, err :=
+			fmtstr.NewTimestampFormatString(&config.Index, staticFields)
+		if err != nil {
+			return nil, err
+		}
+		indexProcessor := add_formatted_index.New(timestampFormat)
+		procs.AddProcessor(indexProcessor)
+	}
+
+	userProcs, err := processors.New(config.Processors)
+	if err != nil {
+		return nil, err
+	}
+	procs.AddProcessors(*userProcs)
+
+	return procs, nil
```

Comment:
Nit: I quite liked the comments you had in the Filebeat implementation, numbering the processor additions and also noting that ordering was significant.

Reply:
Ah, I removed it because metricbeat processor config is simpler (only one configurable processor source), but you're right, an indication of intent here is good so I re-added a comment :-)